src/EloquentRepository.php

```diff
@@ -17,940 +17,953 @@
-				$current_ids = $relation->lists(self::KEY_NAME)->toArray();
-				$new_ids     = array_filter(array_column($input, self::KEY_NAME));
-				$removed_ids = array_diff($current_ids, $new_ids);
-				if (! empty($removed_ids)) {
-					$relation->whereIn(self::KEY_NAME, $removed_ids)->delete();
-				}
-
-				// Set foreign keys on the children from the parent, and save.
-				foreach ($input as $sub_input) {
-					$sub_input[$relation->getPlainForeignKey()] = $parent->{self::KEY_NAME};
-					$relation_repository->setInput($sub_input)->save();
-				}
-				break;
-			case HasOne::class:
-				/**
-				 * @var \Illuminate\Database\Eloquent\Relations\HasOne $relation
-				 */
-				// The parent model "owns" the child model; if we have a new and/or different
-				// existing child model, delete the old one.
-				$current = $relation->getResults();
-				if (! is_null($current)
-					&& (! isset($input[self::KEY_NAME]) || $current->{self::KEY_NAME} !== intval($input[self::KEY_NAME]))
-				) {
-					$relation->delete();
-				}
-
-				// Set foreign key on the child from the parent, and save.
-				$input[$relation->getPlainForeignKey()] = $parent->{self::KEY_NAME};
-				$relation_repository->setInput($input)->save();
-				break;
-			case BelongsToMany::class:
-				/**
-				 * @var \Illuminate\Database\Eloquent\Relations\BelongsToMany $relation
-				 */
-				// Find all the IDs to sync.
-				$ids = [];
-
-				foreach ($input as $sub_input) {
-					$id = $relation_repository->setInput($sub_input)->save()->{self::KEY_NAME};
-
-					// If we were passed pivot data, pass it through accordingly.
-					if (isset($sub_input['pivot'])) {
-						$ids[$id] = (array) $sub_input['pivot'];
-					} else {
-						$ids[] = $id;
-					}
-				}
-
-				// Sync to save pivot table and optional extra data.
-				$relation->sync($ids);
-				break;
-		}
-	}
-
-	/**
-	 * Create a model.
-	 *
-	 * @return \Illuminate\Database\Eloquent\Model
-	 */
-	final public function create()
-	{
-		$model_class = $this->getModelClass();
-		$instance    = new $model_class;
-		$this->fill($instance);
-
-		return $instance;
-	}
-
-	/**
-	 * Read a model.
-	 *
-	 * @return \Illuminate\Database\Eloquent\Model
-	 */
-	final public function read()
-	{
-		return $this->findOrFail($this->getInputId());
-	}
-
-	/**
-	 * Update a model.
-	 *
-	 * @return \Illuminate\Database\Eloquent\Model
-	 */
-	final public function update()
-	{
-		$instance = $this->read();
-		$this->fill($instance);
-
-		// Return the updated instance
-		return $this->read();
-	}
-
-	/**
-	 * Update a model.
-	 *
-	 * @return boolean
-	 */
-	final public function delete()
-	{
-		$instance = $this->read();
-
-		return $instance->delete();
-	}
-
-	/**
-	 * Save a model, regardless of whether or not it is "new".
-	 *
-	 * @return \Illuminate\Database\Eloquent\Model
-	 */
-	final public function save()
-	{
-		$input = $this->getInput();
-
-		return isset($input['id']) ? $this->update() : $this->create();
-	}
+    use ChecksRelations;
```

Comment:
Please change formatting to use tabs instead of spaces

https://www.youtube.com/watch?v=SsoOG6ZeyUI